ui/app/AppLayouts/Browser/views/WebProviderObj.qml

```diff
@@ -30,7 +30,7 @@ QtObject {
     property Connections conn: Connections {
         target: Web3ProviderStore.web3ProviderInst
 
-        onPostMessageResult: {
+        function onPostMessageResult(payloadMethod: string, result: string, chainId: string) {
```

Comment:
For some functions there are "typed" parameters used, for others regular js args. What's the rule for that? Where that js params typing is documented and how it works? :thinking:

Reply:
I have avoided annotations on var types as it's not actually supported by Qt althought the docs say it is . But now that I look on it I missed some annotations on basic types.

Here is the documentation:
https://doc.qt.io/qt-5/qtqml-cppintegration-interactqmlfromcpp.html#invoking-qml-methods

And it became a hot topic now with Qt.6 and new dev on qt quick compiler:
https://www.qt.io/blog/the-new-qtquick-compiler-technology
https://www.qt.io/blog/optimizing-your-qml-application-for-compilation-to-c

Reply:
Generally the feature looks nice, but I'm wondering if it's good to do it here (adding annotations) in a one shot. Can we test it appropriately now (e.g. via forcing compilation to cpp and checking if there is no warnings or other problems)?

Other minor thing - it seems that in some places return type annotations are missing.

Reply:
@micieslak thanks! The benefits of the annotations in our case is really limited. IMO the biggest one in our case is code readability and it was easier to bring them in because this is the nim syntax and usually copy-paste from nim signals did the trick. But if you think there is a chance of misbehaving in our app it might not be worth the risk.

> Can we test it appropriately now (e.g. via forcing compilation to cpp and checking if there is no warnings or other problems)?

Yes, I have compiled them once in another context. Now I'm sorry I didn't save the changes. But I can give it a run again.

> ther minor thing - it seems that in some places return type annotations are missing.

I've just checked those functions and there is not reason to return data. It's really unnatural to return from signal handler.

Reply:
@micieslak Forgot to say that I also tested the signals where the testing steps were clear to me and it works fine. And also the annotation is not breaking the handler even if the argument type does not match the annotation.
But what I'm not confident enough is the nim integration. I've seen too many strange fixes like dummy objects to prevent it from crashing or strange Connections and bindings fixes to make it work properly.

Reply:
@micieslak I've compiled the QMLs and the compile time errors are fixed.